### PR TITLE
Introduce electron-forge

### LINKED ISSUE #36: Introduce to electron-forge
:wave: Hey there! I come to you at the request from @MylesBorins and @jasnell. I've been working with Electron for a while and I'd like to help out with some hard-learned lessons and some best practices.

First up: Build pipeline. Building is one of the hardest things in the Electron world, as there are countless things to consider for each platform, each native dependency, and each build target. We in the Electron maintainer's group have realized that it's the easiest place to get lost, so there's now a common CLI (similar to `ember-cli` or `create-react-app`) that handles the whole build/package/make cycle. Called `electron-forge`, it reduces the common Electron development operations down to simple commands. 

Given that this repo is still in its infancy, this might seem like overkill. However, once you start adding native dependencies and want to build multiple targets (dmg, zip, exe, msi, rpm, deb), you'll be happy that you don't have to worry about the many things required to turn a bundle of JavaScript into an executable :wink:

### PR DESCRIPTION
**Pull request by felixrieseberg**
:wave: Hey there! I come to you at the request from @MylesBorins and @jasnell. I've been working with Electron for a while and I'd like to help out with some hard-learned lessons and some best practices.

First up: Build pipeline. Building is one of the hardest things in the Electron world, as there are countless things to consider for each platform, each native dependency, and each build target. We in the Electron maintainer's group have realized that it's the easiest place to get lost, so there's now a common CLI (similar to `ember-cli` or `create-react-app`) that handles the whole build/package/make cycle. Called `electron-forge`, it reduces the common Electron development operations down to simple commands. 

Given that this repo is still in its infancy, this might seem like overkill. However, once you start adding native dependencies and want to build multiple targets (dmg, zip, exe, msi, rpm, deb), you'll be happy that you don't have to worry about the many things required to turn a bundle of JavaScript into an executable :wink:

This PR introduces `electron-forge`, immediately allowing us to create packages for all systems supported by Electron.

• Added `electron-forge` and basic build pipeline configuration
• Renamed `browser.js` to `renderer.js`. Reason: In the Electron world, the term "browser" refers to Chrome's main process, which is internally called the `browser` process. Given that main and renderer process exhibit different behavior, I took the liberty to rename the file to something more descriptive.

Closes #36

### CODE REVIEW

**Comment by jasnell** [comment]
Rubber-stamp LGTM

**Review by jasnell — APPROVED** [review]

**Review by watilde — APPROVED** [review]
wow nice!

**Review by delvedor — APPROVED** [review]
This is awesome, thank you!
Currently I'm a bit busy at work, I hope I can work again on this project very soon! :)

**Comment by felixrieseberg** [comment]
Nice! So, uhh... how do we merge this?

**Comment by jasnell** [comment]
That would depend on what process folks want to follow for this. To get the issue bootstrapped, I'd recommend using as liberal a process as possible.

@mikeal ... any objections to adding @felixrieseberg as a collaborator on this repo?

File: package.json [hunk 1]
```diff
@@ -15,20 +51,7 @@
     "yo-yo": "^1.2.2"
   },
   "devDependencies": {
-    "electron-builder": "^5.10.5",
-    "electron-prebuilt": "^1.2.5",
+    "electron-prebuilt-compile": "1.6.11",
```
Comment (zeke) [inline package.json:54]: Is this strictly needed?

Comment (anaisbetts) [reply package.json:54]: This is what electron-forge uses to figure out what version of Electron to ship in-box

Comment (felixrieseberg) [reply package.json:54]: Currently, yes. `electron-forge` currently only supports `electron-compile`. That doesn't mean that we need to precompile anything though.

**Comment by delvedor** [comment]
Usually we merge if at least two members approve the pr, we've never say it but usually is how it works.
We should organize ourself a little bit more. If @jasnell  has some advice I'll be very happy to hear it!
However, I'm 👍  on adding @felixrieseberg as collaborator.

**Comment by develar** [comment]
Not clear for what if was required, if electron-builder [was already used](https://github.com/nodejs/installer/pull/17) and "want to build multiple targets" was not a problem at all.

**Pull request merged**